tests/e2e/specs/stories-editor/right-click-menu.js

```diff
@@ -0,0 +1,146 @@
+		expect( page ).toMatchElement( POPOVER_SELECTOR + ' .right-click-remove' );
+		expect( page ).toMatchElement( POPOVER_SELECTOR + ' .right-click-duplicate' );
+	} );
+
+	it( 'does not open the menu by clicking on a page', async () => {
+		const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+		await openRightClickMenu( pageBlock );
+
+		expect( page ).not.toMatchElement( POPOVER_SELECTOR );
+	} );
+
+	it( 'should open right click menu for pasting on a page if a block has been copied previously', async () => {
+		const block = await page.$( BLOCK_SELECTOR );
+		await openRightClickMenu( block );
+
+		await clickButton( 'Copy Block' );
+		const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+		await openRightClickMenu( pageBlock );
+
+		expect( page ).toMatchElement( POPOVER_SELECTOR + ' .right-click-paste' );
+	} );
+
+	it( 'should allow copying and pasting a block', async () => {
+		const block = await page.$( BLOCK_SELECTOR );
+		await openRightClickMenu( block );
+
+		await clickButton( 'Copy Block' );
+
+		await removeAllBlocks();
+		await page.waitForSelector( ACTIVE_PAGE_SELECTOR );
+		const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+		await openRightClickMenu( pageBlock );
+
+		await clickButton( 'Paste' );
+
+		expect( page ).toMatchElement( ACTIVE_PAGE_SELECTOR + ' ' + BLOCK_SELECTOR );
+	} );
+
+	it( 'should allow cutting and pasting a block', async () => {
+		const block = await page.$( BLOCK_SELECTOR );
+		await openRightClickMenu( block );
+
+		await clickButton( 'Cut Block' );
+		expect( page ).not.toMatchElement( BLOCK_SELECTOR );
+
+		const pageBlock = await page.$( ACTIVE_PAGE_SELECTOR );
+		await openRightClickMenu( pageBlock );
+
+		await clickButton( 'Paste' );
+
+		expect( page ).toMatchElement( BLOCK_SELECTOR );
+	} );
+
+	it( 'should allow duplicating a block', async () => {
+		const block = await page.$( BLOCK_SELECTOR );
+		await openRightClickMenu( block );
+
+		await clickButton( 'Duplicate Block' );
+
+		const nodes = await page.$x(
+			'//div[contains(@class, "wp-block-amp-amp-story-post-author")]'
+		);
+		expect( nodes ).toHaveLength( 2 );
+	} );
+
+	it( 'should allow removing a block', async () => {
+		const block = await page.$( BLOCK_SELECTOR );
+		await openRightClickMenu( block );
+
+		await clickButton( 'Remove Block' );
+		expect( page ).not.toMatchElement( BLOCK_SELECTOR );
+	} );
+
+	it( 'should not allow pasting disallowed blocks', async () => {
+		const firstPageClientId = ( await getAllBlocks() )[ 0 ].clientId;
+		await insertBlock( 'Page' );
+		await insertBlock( 'Call to Action' );
+
+		const callToActionSelector = '.wp-block-amp-amp-story-cta';
+		const ctaBlock = await page.waitForSelector( callToActionSelector );
+		await openRightClickMenu( ctaBlock );
+
+		await clickButton( 'Copy Block' );
+
+		await goToPreviousPage();
+		await selectBlockByClientId( firstPageClientId );
+		const pageBlock = await page.$( `#block-${ firstPageClientId }` );
+		// Wait for transition time 300ms.
```

Comment:
Why this a better way of doing this? Waiting around for 300ms feels a little weird.

Reply:
Indeed. Let's find a more suitable indicator, perhaps we can just wait for the `transitionend` event to fire.